### DISCUSSION TITLE
Response templates internationalization

### DISCUSSION THREAD

**Discussion by aaww01** [authelia/authelia#5615] [Ideas]
A couple of questions / comments on how to deal with Internationalization in regards to the email response templates

Authelia does not expose the language of the detected browser to the email response template.  One suggestion would be that the filename (if it exists) IdentityVerification_locale.html is used, failing which IdentityVerification.html is used, or the built-in template as a last resort.  For example, a user accessing with a french browser clicking  reset password would get in order of preference: IdentityVerification_FR.html -> IdentityVerification.html -> built-in template.

Secondly, the {{ .Title }} placeholder is in English only, and the IdentityVerification template makes use of .Title, it would need to be internationalized is some fashion so as to select the appropriate langauge.  Also if the need is to send out bilingual messages, ex: English/French (Canada), English/Spanish (US), etc. This would be a useful feature.

Finally, in the Event.txt example, there is reference to some logic using {{- if ... }} this does not appear to be documented anywhere, but this could perhaps be used in the short term to work around some of the .Title issues.

**Comment by james-d-elliott** [comment]
Hey sorry for not replying sooner.

I have thought a lot about how to do this since we introduced the internationalization for the frontend. I'm partial to the idea of a dictionary we leverage which can be automatically translated similar to how the frontend is translated. This would require the language content to exist in a serializable format like JSON or YAML I believe.

We then just have to include those values as part of the template source, and match it to the browser locale in order and/or the users preference locale (feature to come).
